src/inactive-regions.ts

```diff
@@ -50,26 +50,21 @@ export class InactiveRegionsFeature implements vscodelc.StaticFeature {
     const serverCapabilities: vscodelc.ServerCapabilities&
         {inactiveRegionsProvider?: any} = capabilities;
     if (serverCapabilities.inactiveRegionsProvider) {
-      if (config.get<boolean>('inactiveRegions.useBackgroundHighlight')) {
-        this.decorationType = vscode.window.createTextEditorDecorationType({
-          isWholeLine: true,
-          backgroundColor:
-              new vscode.ThemeColor('clangd.inactiveRegions.background'),
-        });
-      } else {
-        this.decorationType = vscode.window.createTextEditorDecorationType({
-          isWholeLine: true,
-          opacity: config.get<number>('inactiveRegions.opacity').toString()
-        });
-      }
+      this.updateDeclaratedType();
       this.context.subscriptions.push(
           vscode.window.onDidChangeVisibleTextEditors(
               (editors) => editors.forEach(
                   (e) => this.applyHighlights(e.document.fileName))));
       this.context.subscriptions.push(
           vscode.workspace.onDidChangeConfiguration((conf) => {
-            if (!conf.affectsConfiguration('workbench.colorTheme'))
+            const inactiveSettingsChanged =
+              conf.affectsConfiguration('clangd.inactiveRegions.useBackgroundHighlight')
+              || conf.affectsConfiguration('clangd.inactiveRegions.opacity');
```

Comment:
should we check `"clangd.inactiveRegions.background"` as well?

Reply:
I didn't find this option in the settings👀

Reply:
Oh, right, that one is under [colors](https://github.com/clangd/vscode-clangd/blob/6839c067bbcb1a001695048f9087678886e27c46/package.json#L183). So it's probably covered by the `workbench.colorTheme` check then.